tests: net: icmpv4 move to new ztest API

hunterhu3000 opened the pull request:
Move net icmpv4 tests to use new ztest API
1. Add teardown, remove addr
2. Fix a packet release bug in test_send_echo_req_bad_opt
3. Combine send_echo_req/rep, as they should be run in sequence

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

rlubos commented on tests/net/icmpv4/src/main.c:
```diff
@@ -441,12 +451,12 @@ static void test_icmpv4_send_echo_req(void)
-		net_pkt_unref(pkt);
 		zassert_true(false, "Failed to send");
 	}
+	net_pkt_unref(pkt);
```
Why was `net_pkt_unref(pkt);` moved here? If the function returned `NET_OK` here (which is not very clearly verified here) the packet shouldn've been consumed by the stack, unreferencing it here would result in packet being freed twice. Same goes for other places.

hunterhu3000 replied:
You are right. The packet should be released by the lower layer. 
At first I modify the code just as what you have said. But I found the test cannot pass under CONFIG_ZTEST_SHUFFLE=y, the packets failed to be allocated. If I enlarge the packet number in prj.conf. Then it can pass with CONFIG_ZTEST_SHUFFLE=y.
I wonder in the lower layer (or the icmp stack) the packet is not unrefed. 
To WA this problem, I modify the code like this. Maybe we should take it as a bug?

rlubos replied:
The behavior doesn't seem right, the ICMPv4 layer should've released the packet if it notified that the packet was consumed. If that's not the case we should treat this as a bug indeed.

hunterhu3000 commented:
There is a bug in the test, test_send_echo_req_bad_opt() does not release the packet.
```
	if (!net_ipv4_input(pkt)) {
		net_pkt_unref(pkt);
		zassert_true(false, "Failed to send");
	}
```
The bad packet will not be released by lower layer, and it will not released by the test app, due to the "NOT" in the if condition

hunterhu3000 committed:
tests: net: icmpv4 move to new ztest API

Move net icmpv4 tests to use new ztest API
1. Add teardown, remove addr
2. Fix a packet release bug in test_send_echo_req_bad_opt
3. Combine send_echo_req/rep, as they should be run in sequence

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>
